modules/terminate-instances/cloudwatch.tf

```
@@ -0,0 +1,33 @@
+# ----------------------------------------------------------------------------
+# Terminate Instances - CloudWatch/EventBridge Resources
+#
+# This deploys an event rule and target for triggering the provided Lambda
+# function from the ASG lifecycle hook.
+# ----------------------------------------------------------------------------
+resource "aws_cloudwatch_event_rule" "terminate_instances" {
+  name        = "${var.environment}-${var.name}"
+  description = "Trigger GitLab runner instance lifecycle hook on termination."
+
+  event_pattern = <<EOF
+{
+  "source": ["aws.autoscaling"],
+  "detail-type": ["EC2 Instance-terminate Lifecycle Action"],
+  "detail": {
+    "AutoScalingGroupName": ["${var.asg_name}"]
+  }
+}
+EOF
+}
+
+resource "aws_cloudwatch_event_target" "terminate_instances" {
+  rule      = aws_cloudwatch_event_rule.terminate_instances.name
+  target_id = "TriggerTerminateLambda"
```

According the docs this should be a unique identifier, if not set a random will be created. So I suggest either to remove it or add the environment name to make it unique.
```suggestion
  target_id = "${var.environment}-TriggerTerminateLambda"
```